### ISSUE TITLE
Support Multiple fields with type `ID`

### ISSUE THREAD

**Issue by chrisdrackett** [mobxjs/mst-gql#223]
right now we only expect a single ID field. However, in the case a user has more than one ID field on an object mst-gql will not know which one is the ID for the object.

This issue covers updating the config or similar to support letting mst-gql know which is the canonical ID for any model that has more than one ID.

We should provide a clear error during scaffold that we ran into this and give the users steps to fix it.

**Comment by mwarger** [comment]
Something additional to consider: not all schemas use `ID` as the type for ID (Hasura, for example, could have a `uuid` as the ID type).

**Comment by chrisdrackett** [comment]
I'm confused, they use something other than `ID` in the graphQL schema? what is the logic behind this?

**Comment by barbalex** [comment]
@mwarger oh, is that why I always get
```
Uncaught Error: Failed to resolve reference 43e8a390-9447-11ea-8ccb-19e3a3dddd88 to late(function () {
    return _gartenModel__WEBPACK_IMPORTED_MODULE_11__["gartenModel"];
  })
```
when trying to select fields in referenced tables?

My project uses hasura.

**Comment by barbalex** [comment]
related issue: https://github.com/mobxjs/mst-gql/issues/215

**Comment by barbalex** [comment]
Actually there is an issue just about hasura not using type `ID`: https://github.com/mobxjs/mst-gql/issues/92

**Comment by chrisdrackett** [comment]
good point, I feel like we could fix both these issues with a single fix of optimally specifying an ID via config.

/cc @beepsoft

**Comment by Metevier** [comment]
@chrisdrackett 
I am interested in implementing this if it's a feature we feel like is still needed.

How do you feel about the proposal in this comment?
https://github.com/mobxjs/mst-gql/issues/92#issue-493537797

**Comment by chrisdrackett** [comment]
@Metevier I'm all for this! We don't run into this ourselves, so I don't have a lot of ideas around the best way to handle it, but seems like a place where we can add some optional flexibility!
